MockPyth.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "./AbstractPyth.sol";
+import "./PythStructs.sol";
+
+contract MockPyth is AbstractPyth {
+    mapping(bytes32 => PythStructs.PriceFeed) priceFeeds;
+
+    function queryPriceFeed(bytes32 id) public override view returns (PythStructs.PriceFeed memory priceFeed) {
+        return priceFeeds[id];
+    }
+
+    // Takes an array of encoded price feeds and stores them.
+    // You can create this data either by calling createPriceFeedData or 
```

Comment:
in this case, the expected usage would be something like `updatePriceFeeds([createPriceFeedUpdateData(...)])`?

I suggest providing code examples for how to write a unit test in all three different styles.

Reply:
The syntax is different in hardhat/truffle (they treat contracts a bit differently and their test framework is different). I will provide example with one library in the full-stack example that I'm gonna work on.